### ISSUE TITLE
How to Copy Files/Folder from Github Action to Google Cloud Storage (Bucket) 

### ISSUE THREAD

**Issue by MohdRashid01** [google-github-actions/upload-cloud-storage#303]
### TL;DR

I want to push some files or folder from github action to GCP bucket

I have create a new bucket and give permission of owner

BUCKET NAME :- github_action

FOLDER NAME(INSIDE BUCKET I CREATED) :- testing

SOURCE FOLDER-NAME :- ./results.sarif

DEST FOLDER-NAME :- gcp bucket

Whatever above yaml file plse tell me which are i need to kept and which are not to kept(unnecessary content i will remove it)

I Dont know what i need to give here and i dont understand this below 4 point on how to use it in github action and also more link on gcp bucket permission to ready it

path:

destination:

credentials_json: ${{ secrets.gcp_credentials }}

${{ steps.upload-folder.outputs.uploaded }}'

But i want to push this file ./results.sarif into gcp bucket

Please give me any link or blog that clearly show on steps by steps process

### Expected behavior

ERROR 1
Run google-github-actions/upload-cloud-storage@v1

Error: google-github-actions/upload-cloud-storage failed with: ENOENT: no such file or directory, lstat '/home/runner/work/GitHub-Action-Tests/GitHub-Action-Tests/results.sarif'

ERROR 2  
            codeql/upload-sarif action failed: HttpError: Resource not accessible by integration

ERROR 3
        Uploading /home/runner/work/GitHub-Action/test/test.txt to gs://https:/console.cloud.google.com/scanning-vul/test/test.txt
  Error: google-github-actions/upload-cloud-storage failed with: error code invalid_request: Invalid value for "audience". This value should be the full resource name of the Identity Provider. See https://cloud.google.com/iam/docs/reference/sts/rest/v1/TopLevel/token for the list of possible formats.
  node:internal/process/promises:246
            triggerUncaughtException(err, true /* fromPromise */);
            ^
  
  Error: invalid_request
      at Gaxios._request (/home/runner/work/_actions/google-github-actions/upload-cloud-storage/v1/dist/index.js:168:5830)
      at processTicksAndRejections (node:internal/process/task_queues:96:5)
      at async StsCredentials.exchangeToken (/home/runner/work/_actions/google-github-actions/upload-cloud-storage/v1/dist/index.js:168:84570)
      at async IdentityPoolClient.refreshAccessTokenAsync (/home/runner/work/_actions/google-github-actions/upload-cloud-storage/v1/dist/index.js:168:24677)
      at async IdentityPoolClient.getAccessToken (/home/runner/work/_actions/google-github-actions/upload-cloud-storage/v1/dist/index.js:168:23023)
      at async IdentityPoolClient.getRequestHeaders (/home/runner/work/_actions/google-github-actions/upload-cloud-storage/v1/dist/index.js:168:23175)

### Action YAML

```yaml
Here is my gcpauth.yml file

name: Auth
on:
  push:
    branches: [ "main" ]
  workflow_dispatch:
jobs:
  build:
    runs-on: ubuntu-latest
    
    # Add "id-token" with the intended permissions.
    permissions:
      contents: 'read'
      id-token: 'write'
      
    steps:
      # actions/checkout MUST come before auth    
      - uses: 'actions/checkout@v3'
      
      - id: 'auth'
        name: 'Authenticate to Google Cloud'
        uses: 'google-github-actions/auth@v1'
        with:
          workload_identity_provider: 'https://iam.googleapis.com/projects/xxxxxxxxx/locations/global/workloadIdentityPools/workload-auth/providers/oidc/*'
          service_account: 'github@testing-project-xxxxxx.iam.gserviceaccount.com'
          path: ./results.sarif
          destination: github_action/testing/
          
          #credentials_json: ${{ secrets.gcp_credentials }}
          
#       - name: 'Set up Cloud SDK'
#         uses: 'google-github-actions/setup-gcloud@v1'

#       - name: 'Create files'
#         run: |-
#            mkdir -p test
#            touch test/test2.txt

#       - name: 'Use gcloud CLI'
#         run: 'gsutil -m rsync -R test/test2.txt gs://<github_action>'

      
#       - id: 'upload-file'
#         uses: 'google-github-actions/upload-cloud-storage@v1'
#         with:           
#           path: 'https://console.cloud.google.com/storage/browser/github_action'
#           destination: 'github_action/testing'
#         env:
#           files: '${{ steps.upload-folder.outputs.uploaded }}  
```
```#302
